formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -40,42 +40,119 @@ internal class FormulaManagerImpl<Input, State, Output>(
     var transitionID: Long = 0
```

Comment:
These are good points and questions, I will touch on each individual point below. 

I update documentation around `transitionId` (probably better name would be `evaluationId`). 

> Does frame actually have any use for the transitionID member variable? Could we just maintain the single FormulaManagerImpl.transitionId?

Frame is associated with a specific `transitionId` value which allows us to be able to verify if it is still valid or if a new one needs to be created. Without placing it on `SnapshotImpl` or `Frame`, it's hard to know if we are in an out-dated state.

> Does transitionID have any use as a Long? Would it be more understandable as a Boolean? If a boolean is an appropriate use case I wonder if passing it around as argument is necessary.

We might be able to use a boolean, but I found the ID pattern easier to manage. With a boolean, you need to toggle it off in the right place, and it might lead to race conditions if a new event arrives. For example, if someone accidentally calls a listener within `Formula.evaluate`, `transitionId` makes it much easier to catch. In general, we might be able to use `boolean`, but I'm not sure if it will lead to a simpler system.

> I can imagine up scenarios where this might be necessary but not sure if real use cases. Hence some docs could help.

Agreed, I will update the documentation and will rename `transitionId` with `evaluationId` (as I think that better represents why we have it).

> Tangentially related: Should Frame be an inner class or private to the file? It appears to only be used by FormulaManagerImpl. You don't seem to have those elsewhere, but IMO it would help. Formula has many moving parts w/ different roles and if I can easily see that a class is only used locally it helps make digesting the control flow easier.

Yeah, likely it could be a private class within `FormulaManagerImpl`